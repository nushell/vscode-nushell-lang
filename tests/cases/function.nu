export def --env my-function --env [] {}

export def --env "my function1" --env [] {}

export def --env 'my function2' --env [] {}

export def --env  my_function --env [] {}

export def  --env "my_function1" --env [] {}

export def --env 'my_function2' --env [] {}
