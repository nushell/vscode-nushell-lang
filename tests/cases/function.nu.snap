>export def --env my-function --env [] {}
#^^^^^^^^^^^^^^^^ source.nushell entity.name.function.nushell
#                ^ source.nushell
#                 ^^^^^^^^^^^^^^^^^^ source.nushell entity.name.type.nushell entity.name.type.nushell
#                                   ^ source.nushell meta.function.parameters.nushell meta.brace.square.begin.nushell
#                                    ^ source.nushell meta.function.parameters.nushell meta.brace.square.end.nushell
#                                     ^ source.nushell
#                                      ^ source.nushell meta.function.body.nushell punctuation.definition.function.begin.nushell
#                                       ^ source.nushell meta.function.body.nushell punctuation.definition.function.end.nushell
>
>export def --env "my function1" --env [] {}
#^^^^^^^^^^^^^^^^ source.nushell entity.name.function.nushell
#                ^ source.nushell
#                 ^ source.nushell entity.name.type.nushell
#                  ^^^^^^^^^^^^ source.nushell entity.name.type.nushell entity.name.type.nushell
#                              ^ source.nushell entity.name.type.nushell
#                               ^^^^^^ source.nushell entity.name.type.nushell
#                                     ^ source.nushell
#                                      ^ source.nushell meta.function.parameters.nushell meta.brace.square.begin.nushell
#                                       ^ source.nushell meta.function.parameters.nushell meta.brace.square.end.nushell
#                                        ^ source.nushell
#                                         ^ source.nushell meta.function.body.nushell punctuation.definition.function.begin.nushell
#                                          ^ source.nushell meta.function.body.nushell punctuation.definition.function.end.nushell
>
>export def --env 'my function2' --env [] {}
#^^^^^^^^^^^^^^^^ source.nushell entity.name.function.nushell
#                ^ source.nushell
#                 ^ source.nushell entity.name.type.nushell
#                  ^^^^^^^^^^^^ source.nushell entity.name.type.nushell entity.name.type.nushell
#                              ^ source.nushell entity.name.type.nushell
#                               ^^^^^^ source.nushell entity.name.type.nushell
#                                     ^ source.nushell
#                                      ^ source.nushell meta.function.parameters.nushell meta.brace.square.begin.nushell
#                                       ^ source.nushell meta.function.parameters.nushell meta.brace.square.end.nushell
#                                        ^ source.nushell
#                                         ^ source.nushell meta.function.body.nushell punctuation.definition.function.begin.nushell
#                                          ^ source.nushell meta.function.body.nushell punctuation.definition.function.end.nushell
>
>export def --env  my_function --env [] {}
#^^^^^^^^^^^^^^^^ source.nushell entity.name.function.nushell
#                ^^ source.nushell
#                  ^^^^^^^^^^^^^^^^^^ source.nushell entity.name.type.nushell entity.name.type.nushell
#                                    ^ source.nushell meta.function.parameters.nushell meta.brace.square.begin.nushell
#                                     ^ source.nushell meta.function.parameters.nushell meta.brace.square.end.nushell
#                                      ^ source.nushell
#                                       ^ source.nushell meta.function.body.nushell punctuation.definition.function.begin.nushell
#                                        ^ source.nushell meta.function.body.nushell punctuation.definition.function.end.nushell
>
>export def  --env "my_function1" --env [] {}
#^^^^^^^^^^^^^^^^^ source.nushell entity.name.function.nushell
#                 ^ source.nushell
#                  ^ source.nushell entity.name.type.nushell
#                   ^^^^^^^^^^^^ source.nushell entity.name.type.nushell entity.name.type.nushell
#                               ^ source.nushell entity.name.type.nushell
#                                ^^^^^^ source.nushell entity.name.type.nushell
#                                      ^ source.nushell
#                                       ^ source.nushell meta.function.parameters.nushell meta.brace.square.begin.nushell
#                                        ^ source.nushell meta.function.parameters.nushell meta.brace.square.end.nushell
#                                         ^ source.nushell
#                                          ^ source.nushell meta.function.body.nushell punctuation.definition.function.begin.nushell
#                                           ^ source.nushell meta.function.body.nushell punctuation.definition.function.end.nushell
>
>export def --env 'my_function2' --env [] {}
#^^^^^^^^^^^^^^^^ source.nushell entity.name.function.nushell
#                ^ source.nushell
#                 ^ source.nushell entity.name.type.nushell
#                  ^^^^^^^^^^^^ source.nushell entity.name.type.nushell entity.name.type.nushell
#                              ^ source.nushell entity.name.type.nushell
#                               ^^^^^^ source.nushell entity.name.type.nushell
#                                     ^ source.nushell
#                                      ^ source.nushell meta.function.parameters.nushell meta.brace.square.begin.nushell
#                                       ^ source.nushell meta.function.parameters.nushell meta.brace.square.end.nushell
#                                        ^ source.nushell
#                                         ^ source.nushell meta.function.body.nushell punctuation.definition.function.begin.nushell
#                                          ^ source.nushell meta.function.body.nushell punctuation.definition.function.end.nushell
>